arrays/arrays.c

```diff
@@ -83,15 +98,22 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
-
+  if (index > arr->count)
+  {
+    printf("Index not in range.");
+    exit(1);
+  }
   // Resize the array if the number of elements is over capacity
```

Comment:
remember to resize the array if inserting an additional element will make it go over capacity.